plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergStatistics.java

```diff
@@ -221,68 +238,76 @@ public IcebergStatistics build()
         private void updateNullCountStats(int id, Optional<Long> nullCount)
         {
             // If one file is missing nullCounts for a column, invalidate the estimate
-            if (nullCounts.containsKey(id) && nullCounts.get(id).isEmpty()) {
-                return;
-            }
-
-            if (nullCount.isPresent()) {
-                nullCounts.merge(id, nullCount, (existingCount, newCount) -> Optional.of(existingCount.get() + newCount.get()));
-            }
-            else {
-                nullCounts.put(id, Optional.empty());
-            }
+            nullCounts.merge(id, nullCount, (existingCount, newCount) ->
+                    existingCount.isPresent() && newCount.isPresent() ? Optional.of(existingCount.get() + newCount.get()) : Optional.empty());
         }
 
         private void updateMinMaxStats(
                 int id,
                 io.trino.spi.type.Type type,
-                Object lowerBound,
-                Object upperBound,
+                @Nullable Object lowerBound,
+                @Nullable Object upperBound,
                 Optional<Long> nullCount,
                 long recordCount)
         {
             // If this column is only nulls for this file, don't update or invalidate min/max statistics
-            if (nullCount.isEmpty() || nullCount.get() != recordCount) {
-                columnStatistics.putIfAbsent(id, new ColumnStatistics(type, typeManager, lowerBound, upperBound));
-                columnStatistics.get(id).updateMinMax(lowerBound, upperBound);
+            MethodHandle comparisonHandle = typeToComparisonHandle.get(type);
+            if (comparisonHandle != null && (nullCount.isEmpty() || nullCount.get() != recordCount)) {
+                columnStatistics.computeIfAbsent(id, ignored -> new ColumnStatistics(lowerBound, upperBound, comparisonHandle))
+                        .updateMinMax(lowerBound, upperBound);
```

Comment:
i missed that previously -- the `ColumnStatistics` captures initial bounds, so on the first round the call to `updateMinMax` is noop.

You can skip the call in a somewhat verbose manner with `Map.compute`

```java
columnStatistics.compute(id, (ignored, columnStatistics) -> {
    if (columnStatistics == null) {
        columnStatistics = new ColumnStatistics(lowerBound, upperBound, comparisonHandle);
    }
    else {
        columnStatistics.updateMinMax(lowerBound, upperBound);
    }
    return columnStatistics;
});
```

(or document that you're doing what you're doing currently)